lab-matt/lib/power-crud.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const mongoose = require('mongoose');
+mongoose.Promise = require('bluebird');
+const Power = require('../model/power');
+const heroCrud = require('./hero-crud');
+const AppError = require('./app-error');
+const debug = require('debug')('hq:power-crud');
+
+exports.createPower = function(reqBody){
+  return new Promise((resolve, reject)=>{
+    if(! reqBody.heroId) {
+      return reject(AppError.error400('tasks require noteId'));
+    }
+    if(! reqBody.power) {
+      return reject(AppError.error400('powers require a description'));
+    }
+    const power = new Power(reqBody);
+    power.save()
+    .then(power => resolve(power))
+    .catch(err => reject(err));
+  });
+};
+
+exports.deletePower = function(id){
+  debug('hit deletePower');
+  return new Promise((resolve, reject) => {
+    Power.remove({_id: id})
+    .then(resolve)
+    .catch(err => reject(AppError.error404(err.message)));
+  });
+};
+
+exports.fetchPower = function(id){
+  return new Promise((resolve, reject) => {
+    Power.findOne({_id: id})
+    .then(resolve)
+    .catch(err => reject(AppError.error404(err.message)));
+  });
+};
+
+exports.fetchPowerByHeroId = function(heroId){
+  return new Promise((resolve, reject) => {
+    heroCrud.fetchHero({_id: heroId})
```

Comment:
Is this something Duncan wanted you guys to do. I've seen it twice now.
It makes more sense to me to just look up the Power with heroId. Looking up the hero first by their id to then look for a power with their id seems redundant. The only reason I can see is for error catching.